pkg/util/xds/cache.go

```diff
@@ -0,0 +1,407 @@
+// Copyright 2018 Envoyproxy Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package xds
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	envoy_cache "github.com/envoyproxy/go-control-plane/pkg/cache"
+	envoy_log "github.com/envoyproxy/go-control-plane/pkg/log"
+)
+
+// This is a slightly modified version of SnapshotCache from github.com/envoyproxy/go-control-plane
```

Comment:
Can we contribute to go-control-plane?

I've seen pattern before (copying SnapshotCache from x-control-plane) in home grown control plane, because we had to fix something and didn't have time to wait. Then you have to maintain this file - remember to look at changes to this file with every upgrade of go-control-plane.

We can go with it for now, but can we at least make an issue to avoid this?

Reply:
Created a ticket